tests/components/ViewWithCollections.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {View, Text} from 'react-native';
+import _ from 'underscore';
+
+const propTypes = {
+    collections: PropTypes.objectOf(PropTypes.shape({
+        ID: PropTypes.number,
+    })),
+};
+
+const defaultProps = {
+    collections: {},
+};
+
+const ViewWithCollections = (props) => {
+    /* eslint-disable no-console */
+    console.log('rendering ViewWithCollections');
```

Comment:
Really creative! Just thought of an alternative that might be more obvious. We should be able to pass a [`jest.fn()`](https://jestjs.io/docs/en/mock-functions) as a prop and call it in place of this `console.log` the test would then looks something like 

```js
const onRender = jest.fn();
render(<TestComponentWithOnyx onRender={onRender} />); 
expect(onRender.mock.calls.length).toBe(5);
```

Also kind of surprised there isn't a better way to do this with the test renderer.

Reply:
shall try that out